File: templates/Uwp/_composition/MVVMBasic/Project.SplitView/Behaviors/NavigationViewHeaderBehavior_postaction.cs
//{[{
using Param_RootNamespace.Services;
//}]}

namespace Param_RootNamespace.Behaviors
{
    public class NavigationViewHeaderBehavior : Behavior<WinUI.NavigationView>
    {
        protected override void OnAttached()
        {
            //^^
            //{[{
            NavigationService.Navigated += OnNavigated;
            //}]}
        }

        protected override void OnDetaching()
        {
            //^^
            //{[{
            NavigationService.Navigated -= OnNavigated;
            //}]}
        }
    }
}

File: code/TemplateStudioForUWP/Templates/Uwp/_comp/CB/Project.SplitView._VB/Behaviors/NavigationViewHeaderBehavior_postaction.vb
'{[{
Imports Param_RootNamespace.Services
'}]}

Namespace Behaviors
    Public Class NavigationViewHeaderBehavior
        Inherits Behavior(Of WinUI.NavigationView)

        Protected Overrides Sub OnAttached()
'^^
'{[{
            AddHandler NavigationService.Navigated, AddressOf OnNavigated
'}]}
        End Sub

        Protected Overrides Sub OnDetaching()
'^^
'{[{
            RemoveHandler NavigationService.Navigated, AddressOf OnNavigated
'}]}
        End Sub
    End Class
End Namespace
